### PR TITLE
fix(FOROME-1182): table filtration by tags, notes

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-1182

### CODE REVIEW

File: src/pages/ws/ui/control-panel/zone-modals/tags-modal.tsx [hunk 1]
```diff
@@ -15,7 +14,7 @@ export const TagsModal = observer(({ close, title }: ITagsModalProps) => {
   const [searchValue, setSearchValue] = useState('')
 
   useEffect(() => {
-    zoneStore.fetchZoneTagsAsync()
+    zoneStore.tags.length <= 0 && zoneStore.fetchZoneTagsAsync()
```
Comment (AlMaQntr) [inline src/pages/ws/ui/control-panel/zone-modals/tags-modal.tsx]: i think length can't be less than 0. Also, please, use if() { } statement for this

File: src/store/ws/tab-report-paginated.async.store.ts [hunk 2]
```diff
@@ -22,6 +25,50 @@ export class TabReportPaginatedAsyncStore extends BaseAsyncPaginatedDataStore<Ta
     )
   }
 
+  public getPageNo(recNo: number): number | undefined {
+    const currentPage = this.pages.find(page => {
```
Comment (AlMaQntr) [inline src/store/ws/tab-report-paginated.async.store.ts]: use `findIndex` method

File: src/store/ws/tab-report-paginated.async.store.ts [hunk 3]
```diff
@@ -22,6 +25,50 @@ export class TabReportPaginatedAsyncStore extends BaseAsyncPaginatedDataStore<Ta
...
+      }
+    })
+
+    return recIndex ?? undefined
+  }
+
+  public updateRowTags(recNo: number, tags: TTagsDescriptor) {
+    const pageNo = this.getPageNo(recNo)
+
+    if (typeof pageNo === 'number') {
```
Comment (AlMaQntr) [inline src/store/ws/tab-report-paginated.async.store.ts:47]: No need to update row if tags has difference with zones.
I think, you can check it first

**Review by AlMaQntr — APPROVED** [review]

File: src/store/ws/tab-report-paginated.async.store.ts (hunk 3 above)
Comment (Lapk1n) [reply src/store/ws/tab-report-paginated.async.store.ts:47]: Right

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**